pkg/readiness/objset.go

```diff
@@ -27,9 +27,39 @@ type objKey struct {
 	namespacedName types.NamespacedName
 }
 
-type objSet map[objKey]struct{}
-
 func (k objKey) String() string {
 	return fmt.Sprintf("%s [%s]", k.namespacedName.String(), k.gvk.String())
+}
+
+type objData struct {
+	retries int
+}
+
+type objDataMutator func(*objData) *objData
+
+func setRetriesFromFlag(o *objData) *objData {
+	o.retries = *readinessRetries
+	return o
+}
+
+func mutatedObjData(muts ...objDataMutator) objData {
+	out := objData{}
+	for _, m := range muts {
+		m(&out)
+	}
+	return out
+}
+
+type objSet map[objKey]objData
 
+func (o objSet) decrementOrDelete(key objKey) bool {
+	val := o[key]
+	if val.retries > 0 {
+		val.retries--
+		o[key] = val
+		return false
+	} else {
+		delete(o, key)
```

Comment:
Should we just return a recommendation to delete or not so the actual management of all the different maps is handled in one place by the code?